### PR TITLE
Fs 1344 fix tests

### PR DESCRIPTION
**Pull request by srh-sloan**
Converted all unit tests to use a local postgres db with transactional tests. Data is seeded before testing and does not persist afterwards. All tests now runnable individually.

Also added config to use postgres on dev and changed to use common workflow for deployment

### CODE REVIEW

File: .run/Python tests in tests.run.xml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<component name="ProjectRunConfigurationManager">
```
Comment (SentientHousePlant) [inline .run/Python tests in tests.run.xml:1]: 🤔 What is this file?

File: README.md [hunk 2]
```diff
@@ -112,9 +111,26 @@ These are the current pipelines running on the repo:
...
-To run all tests run:
+1. Ensure you have a local postgres instance setup and running with a user `postgres` created.
+1. Install `requirements-dev.txt`
+1. Activate your virtual env: `source .venv/bin/activate`
+1. Run `invoke bootstrap-test-db --database-host=your-db-url` to create a db called "fsd_assess_store_test" in your local postgres instance. This uses the invoke python module to execute tasks defined in `tasks.py`
+1. Run pytest
+- Note you will not see any data inserted from the tests as they run within transactions so are not persisted after the testing session.
 
-    pytest
+## Transactional tests
```
Comment (SentientHousePlant) [inline README.md:121]: 💯 Lovely docs

File: README.md [hunk 3]
```diff
@@ -112,9 +111,26 @@ These are the current pipelines running on the repo:
...
+These rely on the module `pytest-flask-sqlalchemy` which has good docs on its github page: https://github.com/jeancochrane/pytest-flask-sqlalchemy
+
+The main parts of this framework are invoked in `conftest.py` with the following fixture definitions:
+- `enable_transactional_tests` - This makes all tests use transactions so we don't need to turn it on for each test individually
+- `_db` - this makes the framework use our `db` variable from `db.db`, overriding anywhere it is used during the tests.
```
Comment (SentientHousePlant) [inline README.md:126]: 💯 These explanations are great and I will certainly ~~steal~~ add them to the other repos using postgres.

**Review by SentientHousePlant — CHANGES_REQUESTED** [review]
Tiny comments, but overall great work :) 🥇

**Review by harryyo — APPROVED** [review]
Wow! a whole lot of work that substantially improves this repo and it's testing! 👏  Only a couple of minor tidy comments

**Review by harryyo — CHANGES_REQUESTED** [review]
Oh the test_sqlite.db in the root presumably doesn't need to be committed?

File: .run/Python tests in tests.run.xml (hunk 1 above)
Comment (srh-sloan) [reply .run/Python tests in tests.run.xml:1]: Ah this is the pycharm run config, meant to add to gitignore

File: README.md (hunk 3 above)
Comment (srh-sloan) [reply README.md:126]: Thanks guys!

File: tests/test_score_justification_endpoints.py [hunk 4]
```diff
@@ -1,96 +1,95 @@
...
-#             "total_score": 5,
-#             "weight": 0.3,
-#             "weighted_score": 1.5,
-#         },
-#     ]
-#     assessment_id = str(SqliteTestDB.assessment_2_uuid)
-#     endpoint = f"/assessments/{assessment_id}/scores"
-#     response = flask_test_client.get(endpoint)
-#     assert response.json["scores"] == expected_response
+    # @mock.patch('requests.get', side_effect=mock_get_round)
```
[the remark this answers is not in the document]
Comment (srh-sloan) [reply tests/test_score_justification_endpoints.py]: No :)

File: tests/conftest.py [hunk 5]
```diff
@@ -1 +1,189 @@
...
+from colored import fg
+from colored import stylize
+from db import db
+from db.models.assessment import Assessment
+from db.models.compliance import Compliance
+from db.models.criteria import Criteria
+from db.models.scores_justifications import ScoresJustifications
+from db.models.sub_criteria import SubCriteria
+from flask_migrate import upgrade
 from tests.mocks import *  # noqa
```
[the remark this answers is not in the document]
Comment (srh-sloan) [reply tests/conftest.py]: Nope!

File: tests/test_assessment_endpoints.py [hunk 6]
```diff
@@ -1,48 +1,46 @@
...
-        payload = {"applicationId": "abc"}
+        appId = "app_id_1"
+        expected_link_attributes = {"applicationId": appId}
+        payload = {"applicationId": appId}
         endpoint = "/assessments"
-        response = flask_test_client.post(endpoint, json=payload)
+        response = client.post(endpoint, json=payload)
         assessment = response.get_json()
-        self.assessment_ids.append(assessment.get("id"))
+        # self.assessment_ids.append(assessment.get("id"))
```
[the remark this answers is not in the document]
Comment (srh-sloan) [reply tests/test_assessment_endpoints.py]: No

File: .github/workflows/deploy.yml [hunk 7]
```diff
@@ -3,146 +3,52 @@ name: Deploy to Gov PaaS
...
 jobs:
-  testing:
+  test_and_deploy:
+    uses: communitiesuk/funding-design-service-workflows/.github/workflows/deploy.yml@main
+    with:
+      app_name: ${{ github.event.repository.name }}
+      api: true
+      postgres_unit_testing: true
+      db_name: fsd_assess_store_test
+      deploy_to_dev: true
```
Comment (gidsg) [inline .github/workflows/deploy.yml:17]: I think we need to make this `false` as it means we won't be able to iterate at all on migrations in a PR.

Comment (srh-sloan) [reply .github/workflows/deploy.yml:17]: will do in future PR

**Review by SentientHousePlant — APPROVED** [review]

**Review by harryyo — APPROVED** [review]

**Pull request merged**